### PR TITLE
Add remove todo

### PR DESCRIPTION
**Pull request by Hamzaoutdoors**
# To do list: Add & Remove tasks  🎉 

- Add professional Readme file
- fix linters errors
- Set up the project with webpack
- Create HTML dynamically with  HTML webpack Plugins
- Modified structure of webpack for solving issues
- Added basic structure & linters checks
- Added new functionality to add & remove items
- Added function to clear all completed items
- Added arrow functions for best practice of ES6.

## Built With ❤️ 

- HTML & CSS
- JavaScript
- Webpack

## Deployed Preview 🚀 
[Live Preview](https://hamzaoutdoors.github.io/to-do-list/)

### NOTE TO REVIEWER 😉 ❤️

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,36 +1,103 @@
...
+  clearItems.addEventListener('click', () => {
+    tasks = tasks.filter((task) => !task.completed);
+    ul.innerHTML = '';
+    displayItems();
+    IsCompleted.updateLocalStorage(tasks);
+  });
+}
+clearAllCompleted();
+
+// Remove Item from the list
```
Comment (hemantso) [inline src/index.js]: - [ Optional ] Kindly try to delete the comments. Our code should be self-explanatory.

File: src/index.js [hunk 2]
```diff
@@ -1,36 +1,103 @@
...
+
+function remove() {
```
Comment (hemantso) [inline src/index.js]: - [x] As per the learning objective we should follow the ES6 syntax. Kindly try to change the function to the arrow function as per ES6 practice.

File: src/index.js [hunk 3]
```diff
@@ -1,36 +1,103 @@
...
+        innerInput.parentNode.style.backgroundColor = '';
+        innerInput.style.backgroundColor = '';
+      });
+    }
+  });
+}
+
+remove();
+
+function editDesc() {
```
Comment (hemantso) [inline src/index.js]: - [x] As per the learning objective we should follow the ES6 syntax. Kindly try to change the function to the arrow function as per ES6 practice.

File: src/index.js [hunk 4]
```diff
@@ -1,36 +1,103 @@
...
+      event.preventDefault();
+    }
+  });
+  editDesc();
+}
+addToTheList();
+
+// Clear All Completed button
+
+function clearAllCompleted() {
```
Comment (hemantso) [inline src/index.js]: - [x] As per the learning objective we should follow the ES6 syntax. Kindly try to change the function to the arrow function as per ES6 practice.

File: src/index.js [hunk 5]
```diff
@@ -1,36 +1,103 @@
...
+  IsCompleted.changeIcon();
 };
 displayItems();
+
+function clearItems() {
+  const ul = document.getElementById('list');
+  ul.innerHTML = '';
+}
+
+function addToTheList() {
```
Comment (hemantso) [inline src/index.js]: - [x] As per the learning objective we should follow the ES6 syntax. Kindly try to change the function to the arrow function as per ES6 practice.

File: src/index.js [hunk 6]
```diff
@@ -1,36 +1,103 @@
...
     item.index = index;
-    ul.innerHTML += `<li><input type="checkbox" class="checkbox" id="checkbox-${item.index}" ${CHECK}><h4 class="text ${TROUGHLINE}">${item.description}</h4><img class="drop" src="https://img.icons8.com/ios-glyphs/30/000000/menu-2.png" alt="vertical menu"/></li>`;
+    ul.innerHTML += `<li id="${item.index}"><input type="checkbox" class="checkbox" id="checkbox-${item.index}" ${CHECK}><input class="text ${TROUGHLINE} text-${item.index}" type="text" value ="${item.description}"><i class="fa fa-ellipsis-v open" aria-hidden="true"></i><i class="fa fa-trash-o trash d-none" aria-hidden="true"></i></li>`;
   });
   IsCompleted.completeToDo(tasks);
+  IsCompleted.changeIcon();
 };
 displayItems();
+
+function clearItems() {
```
Comment (hemantso) [inline src/index.js]: - [x] As per the learning objective we should follow the ES6 syntax. Kindly try to change the function to the arrow function as per ES6 practice.

File: src/index.js [hunk 7]
```diff
@@ -1,36 +1,103 @@
...
-  description: 'Meet friends',
-  completed: false,
-  index: 3,
-},
-];
+let tasks = JSON.parse(localStorage.getItem('ToDo')) || [];
+
+// Display Items in the UI
 
 const displayItems = () => {
```
Comment (hemantso) [inline src/index.js:7]: - [x] As per the learning objective we should follow the ES6 syntax. Kindly try to change the function to the arrow function as per ES6 practice.

**Review by hemantso — CHANGES_REQUESTED** [review]
Hi @Hamzaoutdoors , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there! 

Alt Text
## Required Changes ♻️
_Check the comments under the review._

Cheers and happy coding 😃

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. You can mention me (@hemant-soni-vst-au4) so I can see your comments and respond.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 2 above)
Comment (Hamzaoutdoors) [reply src/index.js]: Done 👍 .. thanks for your feedback

File: src/index.js (hunk 3 above)
Comment (Hamzaoutdoors) [reply src/index.js]: Done 👍 .. thanks for your feedback

File: src/index.js (hunk 5 above)
Comment (Hamzaoutdoors) [reply src/index.js]: Done 👍 .. thanks for your feedback

File: src/index.js (hunk 6 above)
Comment (Hamzaoutdoors) [reply src/index.js]: Done 👍 .. thanks for your feedback

File: src/index.js (hunk 7 above)
Comment (Hamzaoutdoors) [reply src/index.js:7]: Done 👍 .. thanks for your feedback

File: src/index.js (hunk 4 above)
Comment (Hamzaoutdoors) [reply src/index.js]: Done 👍 .. thanks for your feedback

**Review by descholar-ceo — APPROVED** [review]
Hi @Hamzaoutdoors , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**